packages/bitcore-node/src/models/transaction.ts

```diff
@@ -274,49 +282,179 @@ export class TransactionModel extends BaseTransaction<IBtcTransaction> {
-        result[0].name = name;
-        result[0].decimals = decimals;
-        result[0].symbol = symbol;
-        result[0].totalSupply = totalSupply;
+      }
+      if (
+        result.length > 0 &&
+        result[0].log.length > 0 &&
+        result[0].log[0].topics.length > 0 &&
+        result[0].log[0].topics[0] === 'ddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef'
+      ) {
+        result[0].events = [];
+        const receipt = result[0];
+        const from = receipt.log[0].topics[1].replace('000000000000000000000000', '');
+        const to = receipt.log[0].topics[2].replace('000000000000000000000000', '');
+        const value = parseInt(receipt.log[0].data, 16);
+        const contractAddress = receipt.log[0].address;
+        const token = await TokenStorage.collection.findOne({ chain, network, contractAddress });
+        const fromAddress = await fromHexAddress({ address: from, chain, network });
+        const toAddress = await fromHexAddress({ address: to, chain, network });
+        const balanceFrom = await rpc.call('frc20balanceof', [contractAddress, fromAddress]);
+        const balanceTo = await rpc.call('frc20balanceof', [contractAddress, toAddress]);
+        const decimals = token ? token.decimals : await rpc.call('frc20decimals', [contractAddress]);
+        const balanceFromDecimal = countDecimals(+balanceFrom);
+        const balanceToDecimal = countDecimals(+balanceTo);
+        TokenBalanceStorage.collection.updateOne(
+          { contractAddress, address: from },
+          {
+            $set: {
+              chain,
+              network,
+              contractAddress,
+              address: from,
+              balance: (
+                BigInt(new Decimal(balanceFrom).mul(new Decimal(10 ** balanceFromDecimal)).toString()) *
+                BigInt(10 ** (decimals - balanceFromDecimal))
+              ).toString(),
+            },
+          },
+          { upsert: true }
+        );
+        TokenBalanceStorage.collection.updateOne(
+          { contractAddress, address: to },
+          {
+            $set: {
+              chain,
+              network,
+              contractAddress,
+              address: to,
+              balance: (
+                BigInt(new Decimal(balanceTo).mul(new Decimal(10 ** balanceToDecimal)).toString()) *
+                BigInt(10 ** (decimals - balanceToDecimal))
+              ).toString(),
+            },
+          },
+          { upsert: true }
+        );
+        result[0].events.push({
+          type: 'transfer',
+          from,
+          to,
+          value,
+        });
       }
       this.collection.updateOne(
         { txid, chain, network },
         {
           $set: {
-            receipt: result
-          }
+            receipt: result,
+          },
         }
       );
     } catch (err) {
       console.error({ chain, network, txid });
       console.error(err);
+      await new Promise((f) => setTimeout(f, 5000));
```

Comment:
what happen if we just throw error out?

Reply:
It throws an error and restarts docker, but it can sync correctly kub. I will change it to throw error kub.